### ISSUE TITLE
userinfo.db error again

### ISSUE THREAD

**Issue by asmadeus08** [cbeuw/Cloak#72]
The same problem as it was described here https://github.com/cbeuw/Cloak/issues/13 happened to me too. But advice doesn't help at all. Try to use the same config and Ubuntu version the same but error still appears.

**Comment by beanslel** [comment]
Did you try making a folder like /home/public, set that folder as databasepath and then set permissions to 777 for that folder? That worked for me.

**Comment by asmadeus08** [comment]
> Did you try making a folder like /home/public, set that folder as databasepath and then set permissions to 777 for that folder? That worked for me.

Of course, I try. I read previous issue but getting the same error. Any idea? Does it matter on which one Ubuntu version run this script https://github.com/HirbodBehnam/Shadowsocks-Cloak-Installer? Did you use that?

**Comment by HirbodBehnam** [comment]
Hello
My script works on Ubuntu 16 and newer.
Also my script runs the Cloak as root and puts the database at `/etc/cloak/userinfo.db`. The folder is created is accessible by root.

If you want to test the database, temporary, use `/tmp/` folder.

And also, the cloak port and shadowsocks port must not be same.

**Comment by asmadeus08** [comment]
> Hello
> My script works on Ubuntu 16 and newer.
> Also my script runs the Cloak as root and puts the database at `/etc/cloak/userinfo.db`. The folder is created is accessible by root.
> 
> If you want to test the database, temporary, use `/tmp/` folder.
> 
> And also, the cloak port and shadowsocks port must not be same.

Ok, so I  did the same on Ubuntu 16.04 before but it not works. And I got error with userinfo.db as described. I used 443 port on Cloak and on shadowsocks-libev with 995 port. So where I'm wrong? I'm even try to change folder of userinfo.db but still the same error. Any advice?

**Comment by HirbodBehnam** [comment]
Can you try `/tmp`? It's not a good idea you can just test that if it works there.

**Comment by WordlessEcho** [comment]
Set path to `/tmp` is work for me. Before that, I set path to `/etc/ck-server` and already set its permission to 777. But it does not work.

**Comment by klaaktu** [comment]
Sorry for digging up the post.
I see that shadowsocks-libev has `DynamicUser=true` in its systemd service file, which prevents storing persistent file.
Maybe this is what's causing the problem with userinfo.db when using Cloak as a plugin for SS?

**Comment by malikshi** [comment]
I got errors often. Even when the config all good but the log says the UID is incorrect. It's troublesome when u has many users to generate and distributed again

**Comment by lessload** [comment]
Permission problem. may cause security problem in the future.
